sqlglot/dialects/tsql.py

```diff
@@ -143,9 +166,8 @@ class Parser(Parser):
             **Parser.FUNCTIONS,
             "CHARINDEX": exp.StrPosition.from_arg_list,
             "ISNULL": exp.Coalesce.from_arg_list,
-            "DATEADD": lambda args: exp.DateAdd(
-                this=list_get(args, 2), expression=list_get(args, 1), unit=list_get(args, 0)
-            ),
+            "DATEADD": lambda args: parse_date_delta(args, kind="ADD"),
+            "DATEDIFF": lambda args: parse_date_delta(args, kind="SUB"),
```

Comment:
We can further simplify by just passing in the expression class directly here, instead of comparing strings.

Reply:
Also, there are some similar examples in bigquery, mysql (_date_add) you can check out.